src/shared/pretty-print.c

```diff
@@ -160,6 +160,24 @@ static int cat_file(const char *filename, bool newline) {
                 if (r == 0)
                         break;
 
+                if (flags & CAT_FLAGS_SUPPRESS_COMMENTS) {
+                        if (line[0] == '[') {
+                                /* The start of a section, let's not print it yet. */
+                                free_and_replace(section, line);
+                                continue;
```

Comment:
i think if you have multiple identical section names one after the other (which happens when drop-ins are coalesced) then we should also suppress all but one

Reply:
The file name is printed for each drop-in. So I think we should print the section header too. Current behaviour:
```
# a.conf
[section]
setting = value

# b.conf
[section]
setting = value
```
If redundant sections were dropped:
```
# a.conf
[section]
setting = value

# b.conf
setting = value
```
Seems iffy.

Reply:
well, i was also thinking of cases where people generate unit files with some scripting language and get tons of redundant `[foobar]` in the same unit file. 

i agree it makes sense to keep it given the filename comment, but i think *within* a file we should still try to suppress redundant section headers.